mist/runtime/loss.py

```diff
@@ -31,6 +31,64 @@ def forward(self, y_true, y_pred):
         loss = torch.mean(loss)
         return loss
 
+#Implement clDice:
+class soft_cldice(nn.Module):
+    def __init__(self, iter_=3, smooth = 1., exclude_background=False):
+        super(soft_cldice, self).__init__()
+        self.iter = iter_
+        self.smooth = smooth
+        self.soft_skeletonize = SoftSkeletonize(num_iter=10)
+        self.exclude_background = exclude_background
+
+    def forward(self, y_true, y_pred):
+        if self.exclude_background:
+            y_true = y_true[:, 1:, :, :]
+            y_pred = y_pred[:, 1:, :, :]
+        skel_pred = self.soft_skeletonize(y_pred)
+        skel_true = self.soft_skeletonize(y_true)
+        tprec = (torch.sum(torch.multiply(skel_pred, y_true))+self.smooth)/(torch.sum(skel_pred)+self.smooth)    
+        tsens = (torch.sum(torch.multiply(skel_true, y_pred))+self.smooth)/(torch.sum(skel_true)+self.smooth)    
+        cl_dice = 1.- 2.0*(tprec*tsens)/(tprec+tsens)
+        return cl_dice
+
+
+def soft_dice(y_true, y_pred):
```

Comment:
This is already implemented in the same file:

https://github.com/mist-medical/MIST/blob/41c6fef4aa3d4d4418023d56e4066c0d227667bd/mist/runtime/loss.py#L15

Please see if you can use this implementation. We don't want to reimplement the same functions.